sys/kern/time.c

```diff
@@ -17,9 +19,104 @@ int do_clock_gettime(clockid_t clk, timespec_t *tp) {
   return 0;
 }
 
-int do_clock_nanosleep(clockid_t clk, int flags, const timespec_t *rqtp,
+systime_t ts2hz(timespec_t *ts) {
+  systime_t ticks;
+
+  if (ts->tv_sec < 0 || (ts->tv_sec == 0 && ts->tv_nsec == 0))
+    return 0;
+
+  if (ts->tv_sec <= UINT_MAX / CLK_TCK) {
+    const int pow9 = 1000000000;
+    int tick = pow9 / CLK_TCK;
+
+    /* We are rounding up the number of ticks */
+    long nsectck = (ts->tv_nsec + tick - 1) / tick;
+    ticks = ts->tv_sec * CLK_TCK;
+
+    if (ticks <= UINT_MAX - nsectck - 1) {
+      /* We are adding 1 for the current tick to expire */
+      ticks += nsectck + 1;
+    } else
+      ticks = UINT_MAX;
+
+  } else
+    ticks = UINT_MAX;
```

Comment:
I'm not sure what our C style guide says here `if { ... } else ...`. Consult it with @cahirwpz

Reply:
Thank you, when we have if-else statements at the same time we want to use brackets for both of them or none. Done.